### PR TITLE
network: add RSSI report App

### PR DESCRIPTION
**Pull request by pillip8282**
add RSSI report App
TASH> rreport ssid password url
then it send rssi info to the server every 500ms

### CODE REVIEW

File: apps/examples/rssi_report/rreport_http.c [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
```
Comment (jeongchanKim) [inline apps/examples/rssi_report/rreport_http.c:19]: I think we don't need this.

File: apps/examples/rssi_report/rreport_http.c [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+		RR_ERR;
+		return -1;
+	}
+
+	printf("url: %s str:%s \n", url, str);
+
+	CURLcode ret = curl_easy_setopt(ctx, CURLOPT_URL, url);
+	if (ret != CURLE_OK) {
+		RR_ERR;
+		return -1;
```
Comment (jeongchanKim) [inline apps/examples/rssi_report/rreport_http.c:40]: Is it okay not to call the `curl_easy_cleanup`?

File: apps/examples/rssi_report/rreport_utils.c [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	nowtime = tv.tv_sec + 9 * 3600; // add additional number to support timezon
+	nowtm = localtime(&nowtime);
+	strftime(tmbuf, sizeof tmbuf, "%Y-%m-%d %H:%M:%S", nowtm);
+	snprintf(buf, size, "%s.%06ld", tmbuf, tv.tv_usec);
+
+	return 0;
+}
+
+
+default:
```
Comment (jeongchanKim) [inline apps/examples/rssi_report/rreport_utils.c:66]: Could you check that this is needed?

File: apps/examples/rssi_report/rreport_utils.c [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+int rr_get_time(char *buf, int size)
+{
+	struct timeval tv;
+	time_t nowtime;
+	struct tm *nowtm;
+	char tmbuf[64];
+
+	gettimeofday(&tv, NULL);
+	nowtime = tv.tv_sec + 9 * 3600; // add additional number to support timezon
```
Comment (jeongchanKim) [inline apps/examples/rssi_report/rreport_utils.c:57]: Please change `timezon` to `timezone`.

File: apps/examples/rssi_report/rreport_http.c (hunk 1 above)
Comment (pillip8282) [reply apps/examples/rssi_report/rreport_http.c:19]: could you elaborate it? I can't understand.

Comment (jeongchanKim) [reply apps/examples/rssi_report/rreport_http.c:19]: `<tinyara/config.h>` is needed only if there is `CONFIG_XXX` in this file. 
But I think there is no in this file.

Comment (pillip8282) [reply apps/examples/rssi_report/rreport_http.c:19]: curl.h contains CONFIG_XXX so adding tinyara/config.h is safer.

File: apps/examples/rssi_report/rreport.c [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+		char *msg = rr_generate_message(rssi, RR_NUM_OF_RSSI);
+		if (!msg) {
+			RR_ERR;
+			continue;
+		}
+
+		rr_send_result(msg, app_msg->url);
+
+		free(msg);
+	}
```
[the remark this answers is not in the document]
Comment (pillip8282) [reply apps/examples/rssi_report/rreport.c:73]: it supposed to test infinitely. you have to reset board to finish it.

**Pull request merged**